recipes/qxmpp/all/conanfile.py

```diff
@@ -27,62 +33,90 @@ class QxmppConan(ConanFile):
         "with_gstreamer": False,
     }
 
-    generators = "cmake", "cmake_find_package_multi"
+    @property
+    def _min_cppstd(self):
+        return 17
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "10",
+            "Visual Studio": "17",
+            "msvc": "192",
+            "clang": "8",
+            "apple-clang": "13",
+        }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("qt/6.2.4")
+        self.requires("qt/6.6.1", transitive_headers=True, transitive_libs=True)
         if self.options.with_gstreamer:
-            self.requires("gstreamer/1.19.2")
-            self.requires("glib/2.70.1")
+            self.requires("gstreamer/1.22.3")
+            self.requires("glib/2.78.3")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def build_requirements(self):
+        self.tool_requires("qt/<host_version>")
```

Comment:
Could you please explain is Qt needed as a build requirement? Please, in case needing it, show a log or link indicating it.